apps/versions/views.py

```diff
@@ -110,12 +110,16 @@ def guard():
 
 
 @addon_view_factory(guard)
-def download_latest(request, addon, type='xpi', platform=None):
+def download_latest(request, addon, beta=False, type='xpi', platform=None):
     platforms = [amo.PLATFORM_ALL.id]
     if platform is not None and int(platform) in amo.PLATFORMS:
         platforms.append(int(platform))
+    if beta and addon.show_beta:
```

Comment:
What happens if there's a `-beta` in the url, but `show_beta` is False (because for example there's no beta)? We'll end up serving a version that isn't beta, right? Shouldn't we instead return a 404?